From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Grische <2787581+grische@users.noreply.github.com>
Date: Thu, 9 Apr 2026 23:41:55 +0200
Subject: [PATCH 8/9] interface: skip reset on ubus timeout

Triggers when node-whisperer starts early in the system.

(cherry picked from commit 1c413398f4b80b15a71ad9cfa982327c6c48b7df)
---
 src/interface.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/interface.c b/src/interface.c
index 0091635..354251f 100644
--- a/src/interface.c
+++ b/src/interface.c
@@ -67,6 +67,10 @@ int nw_interface_update(struct ubus_context *ctx, char *vendor_elements)
 
 		log_debug("Sending vendor elements to id=%d name=%s", iface->ubus.id, iface->ubus.name);
 		ret = ubus_invoke(ctx, iface->ubus.id, "set_vendor_elements", b.head, NULL, NULL, 1000);
+		if (ret == UBUS_STATUS_TIMEOUT) {
+			log_info("Timeout sending vendor elements to id=%d name=%s", iface->ubus.id, iface->ubus.name);
+			continue;
+		}
 		if (ret) {
 			log_error("Failed to send vendor elements to id=%d name=%s error=%s", iface->ubus.id, iface->ubus.name, ubus_strerror(ret));
 
-- 
2.53.0

